src/components/DownloadButton.js

```diff
@@ -1,47 +1,81 @@
 // Download button for individual plots
+import { saveAs } from "file-saver";
 import React from "react";
 import { Button } from "react-bootstrap";
 import "./DownloadButton.css";
-// import JSZip from "jszip";
 
-// import Nesting from "./ImageSaving";
+import JSZip from "jszip";
+import JSZipUtils from "jszip-utils";
 
-function DownloadButton() {
-  // For multiple downloads
-  //   var zip = new JSZip();
-  //   var count = 0;
-  //   var zipFilename = "UKCORDEX-plots.zip";
-  //   var urls = ["http://image-url-1", "http://image-url-2", "http://image-url-3"];
+function DownloadButton({ plotvars, seasons, periods, plottypes, regions }) {
+  // zip multiple images
+  const downloadzip = (imglist) => {
+    var zip = new JSZip();
+    var count = 0;
+    var zipFilename = "UKCORDEX-plots.zip";
+    var urls = imglist;
 
-  //   urls.forEach(function (url) {
-  //     var filename = "filename";
-  //     // loading a file and add it in a zip file
-  //     JSZipUtils.getBinaryContent(url, function (err, data) {
-  //       if (err) {
-  //         throw err; // or handle the error
-  //       }
-  //       zip.file(filename, data, { binary: true });
-  //       count++;
-  //       if (count == urls.length) {
-  //         zip.generateAsync({ type: "blob" }).then(function (content) {
-  //           saveAs(content, zipFilename);
-  //         });
-  //       }
-  //     });
-  //   });
+    urls.map((url) => {
+      // get the right basename for each image
+      var filename = url.slice(url.lastIndexOf("/") + 1);
+      // loading a file and add it in a zip file
+      JSZipUtils.getBinaryContent(url, function (err, data) {
+        if (err) {
+          throw err; // or handle the error
+        }
+        zip.file(filename, data, { binary: true });
+        count++;
+        if (count === urls.length) {
+          zip.generateAsync({ type: "blob" }).then(function (content) {
+            saveAs(content, zipFilename);
+          });
+        }
+      });
+      return console.log("this needed a return");
+    });
+  };
+  // Handles what happens when save plot button is clicked
+  const handleClick = () => {
+    let imglist = [];
+    plotvars.map((pvar) =>
+      seasons.map((season) =>
+        periods.map((period) =>
+          plottypes.map((ptype) => {
+            // TODO: add region
+            // regions.map((region) => {
```

Comment:
This can be uncommented now we are passing `regions` to the component.

Reply:
We'll leave it commented to keep github happy until we have files with the region